python/lsst/meas/astrom/exceptions.py

```diff
@@ -0,0 +1,79 @@
+# This file is part of meas_astrom.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AstrometryError", "AstrometryFitFailure", "BadAstrometryFit", "MatcherFailure"]
+
+import lsst.pipe.base
+
+
+class AstrometryError(lsst.pipe.base.AlgorithmError):
+    """Parent class for failures in astrometric fitting.
+
+    Parameters
+    ----------
+    msg : `str`
+        Informative message about the nature of the error.
+    kwargs
+        All other arguments are added to a ``_metadata`` attribute, which is
```

Comment:
I'm torn on that: I directly modify `_metadata` in AstrometryTask to add more information to exceptions that came from the matcher or fitter. I could name it something non-"private", but `metadata` (the read-only property) is the most obvious, but taken, choice.

Reply:
I think I see that it's effectively "protected" rather than just "private", and those are always in a nebulous state in Python already.  Fine to leave it as-is.